_overviews/overview/a-taste-of-scala.md

```diff
@@ -0,0 +1,1329 @@
+
+<!-- TODO: links -->
+`for` loops and expressions are covered in more detail in the Control Structures sections of this Overview and in the Reference documentation.
+
+
+### match expressions
+
+Scala has a `match` expression, which in its most basic use is like a Java `switch` statement:
+
+```scala
+val i = 1
+
+// later in the code ...
+i match
+  case 1 => println("one")
+  case 2 => println("two")
+  case _ => println("other")
+```
+
+However, a `match` expression really is an expression, meaning that it returns a result based on the pattern match:
+
+```scala
+val result = i match
+  case 1 => "one"
+  case 2 => "two"
+  case _ => "other"
+```
+
+The `match` expression isn’t limited to just integers, it can be used with any data type, including booleans:
+
+```scala
+val booleanAsString = bool match
+  case true => "true"
+  case false => "false"
+```
+
+In fact, a `match` expression can be used to test a variable against multiple patterns. This example shows (a) how to use a `match` expression as the body of a method, and (b) how to match all the different types shown:
+
+<!-- TODO use scalafiddle -->
+```scala
+def getClassAsString(x: Any): String = x match
+  case s: String => s"'$s' is a String"
+  case i: Int => "Int"
+  case d: Double => "Double"
+  case l: List[_] => "List"
+  case _ => "Unknown"
+
+// examples
+getClassAsString(1)             // Int
+getClassAsString("hello")       // 'hello' is a String
+getClassAsString(List(1,2,3))   // List
+```
+
+Pattern matches can get as complicated as you need, and you’ll see more examples of it in the Control Structures sections of this Overview and in the Reference documentation.
+
+
+### try/catch/finally
+
+Scala’s `try`/`catch` control structure lets you catch exceptions. It’s similar to Java, but its syntax is consistent with `match` expressions:
+
+<!-- TODO: better example; show 'finally' -->
+```scala
+try
+  writeToFile(text)
+catch
+  case fnfe: FileNotFoundException => println(fnfe)
+  case ioe: IOException => println(ioe)
+```
+
+
+### while loops
+
+<!-- TODO: use parens and braces? -->
+
+Scala also has a `while` loop construct. It’s one-line syntax looks like this:
+
+```scala
+while x >= 0 do x = f(x)
+```
+
+And it’s multiline syntax looks like this:
+
+```scala
+var x = 1
+
+// parentheses
+while (x < 3)
+  println(x)
+  x += 1
+
+// no parens
+while
+  x < 3
+do
+  println(x)
+  x += 1
+```
+
+### Create your own control structures!
+
+In Scala you can also create code that works just like a control structure. You can learn more about this in the Control Structures chapter in the Reference documentation.
+<!--
+NOTE: my old examples are `whilst` and `doubleIf`; `using` is a good example.
+-->
+
+
+
+## Data Modeling
+
+Scala supports both functional programming (FP) and object-oriented programming (OOP), as well as a fusion of the two paradigms. This section provides a quick overview of data modeling in OOP and FP.
+
+
+### OOP data modeling (traits and classes)
```

Comment:
I might be nitpicky but for me in OOP we don’t really model _data_. It’s more about modeling operations, imho.

Reply:
I’ll update this.